crates/goose/examples/canonical_model_checker.rs

```diff
@@ -0,0 +1,369 @@
+/// Canonical Model Checker
```

Comment:
should this be in examples?

in general how are we going to integrate this with the overall process. how long does it take to run all of this? can we add it to the CI and then fail when it is not up to date? or add it to the release process /cc @jamaedo